### ISSUE TITLE
Application is not compatible with python 3.10

### ISSUE THREAD

**Issue by SuperFoo42** [hhu-rapp/rapp-tool#1]
When following the README, the application doesn't work with python 3.10, as the package versions are not compatible with the python version, which leads to all sorts of interesting errors when installing the packages.
However, I solved this problem with a simple update of the packages. Now, the application starts, but is still not functional, because in this case, the following happens:

When running the application under python 3.10, the GUI freezes on the training example, when starting the training. Subsequently, the application crashes with the following error:

`Traceback (most recent call last):
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/pipeline.py", line 243, in train
    self.qmainwindow.tabs.widget(self.qmainwindow.evaluation_tab_index).populate_tabs(pl,
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/evaluation.py", line 137, in populate_tabs
    self._populate_inspection_tab(pipeline)
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/evaluation.py", line 273, in _populate_inspection_tab
    self._populate_inspection_table(pipeline.data, pipeline.sensitive_attributes, pipeline.performance_results,
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/evaluation.py", line 286, in _populate_inspection_table
    self.inspectionPerformanceTable = InspectionPerformanceTable(data, model, performance_results)
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/widgets/__init__.py", line 375, in __init__
    self.cm_groupBox[mode] = ConfusionMatrixTable(confusion_matrix)
  File "/home/paul/Projekte/BTW_115/115/rapp-tool/rapp/gui/widgets/__init__.py", line 417, in __init__
    tableGridLayout.addWidget(labelTitle, 0, 1 + len(confusion_matrix) / 2, 1, len(confusion_matrix),
TypeError: arguments did not match any overloaded call:
  addWidget(self, QWidget): too many arguments
  addWidget(self, QWidget, int, int, alignment: Union[Qt.Alignment, Qt.AlignmentFlag] = Qt.Alignment()): argument 3 has unexpected type 'float'
  addWidget(self, QWidget, int, int, int, int, alignment: Union[Qt.Alignment, Qt.AlignmentFlag] = Qt.Alignment()): argument 3 has unexpected type 'float'
[1]    525400 IOT instruction (core dumped)  python -m rapp.gui`

When running the application with python 3.8, the GUI still freezes, but comes back alive, once the training is finished.

**Comment by jdnklau** [comment]
Thanks for reporting the issue!

We updated the requirements.txt and adjusted the code accordingly. If you have time to spare, would you be so kind as to double check whether the crash still happens for you?

Regarding the UI-freeze during training: As for now, this is expected behaviour as the training does not happen in a separate thread, but we will work on that soon.

**Comment by SuperFoo42** [comment]
Still the same problem. With python 3.8 it works, but on python3.10 it crashes after generating the report with the same error message.
Reading from the error message, it seems to be a problem with the Qt API that might have changed.

**Comment by jdnklau** [comment]
Thank you, @SuperFoo42 

We will look into it more closely and I'll get back at you.

**Comment by jdnklau** [comment]
Hi SuperFoo42, thank you for your patience and time.

We fixed an issue with a segmentation fault occurring after training, but I was not able to reproduce your error message on Python 3.10.10. 

Going over your error message listed above, I see the crash happens in `rapp/gui/widgets/__init__.py`
on line 417:
```python
tableGridLayout.addWidget(labelTitle, 0, 1 + len(confusion_matrix) / 2, 1, len(confusion_matrix),
                          alignment=Qt.AlignCenter)
```

This error might be related to one we fixed 4 weeks ago in this commit: 76a4b2936330643e474f69b8412d9806640deae3.

Is that still the error you got a week ago?

If you have the time, I'd ask you to verify that you are on 653376b0a334cdb1ae408150f5a9497462a73bbe for both Python versions and to check if you still are experiencing any crashes after training. If you still receive errors on Python 3.10, could you please be so kind to post them again?

Thank you again so much for your help in locating these issues!

**Comment by SuperFoo42** [comment]
Hi jdnklau,

with 653376b everything works, no more segfaults.  I saw 76a4b29 and thought I checked it out to test it but back then it didn't work. I think I was on the wrong commit, sorry, my bad. 
Thank you for fixing.
